### ISSUE TITLE
building a second time while first build runs makes building fail permanently

### ISSUE THREAD

**Issue by karyon** [thomasjo/atom-latex#250]
using pdfTeX, Version 3.14159265-2.6-1.40.17 (MiKTeX 2.9 64-bit), on windows 10

To reproduce:
1. press ctrl-alt-b
2. while it's building, press it again
3. compilation will fail.
4. whenever trying to build again, it will fail after half a second, without writing anything to the log.
5. delete the .aux file, compile again, it works now.

when sumatraPDF has the output opened already, one needs to additionally close sumatraPDF and delete the output pdf file.

**Comment by yitzchak** [comment]
Are you using `texify` or `latexmk` and which version of `atom-latex` are you on?

**Comment by karyon** [comment]
latexmk and 0.37.1

**Comment by yitzchak** [comment]
Did you get any error messages in the LaTeX log in the status bar?

**Comment by karyon** [comment]
in step 3 it says "Latex compilation error", and the log gives this here
```
<filename>.aux:20: Text line contains an invalid character
A funny symbol that I can't read has just been input.
Continue, and I'll forget that it ever happened.
```
a hundred times before aborting.

**Comment by karyon** [comment]
this is the .aux file after step 3:
```
\relax 
\citation{Keller:1997:InstantRadiosity}
\citation{Dachsbacher:2005:RSM}
\citation{ritschel2011ismsViewAdaptive}
\citation{prutkin2012reflective}
\citation{hedman2016sequential}
\citation{ritschel2008ism}
\citation{evans2015dreams}
\citation{ritschel2011ismsViewAdaptive}
\citation{hollander2011manylods}
\citation{barak2013temporally}
\citation{ritschel2008ism}
\citation{barak2013temporally}
\citation{sugihara2014layered}
\citation{sun2015manylightsSVO}
\citation{Chen:2016:Compactvoxels}
\citation{segovia2006bidirectional}
\citation{novak2011screen}
\citation{tokuyoshi2015vsgl}
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               thepage }}
\@writefile{lof}{\contentsline {figure}{\numberline {3}{\ignorespaces Light leaks caused by the imperfect shadow maps rendered with the splat renderer (left) and single-pixel renderer (right). VPLs from beneath the curtains leak light onto the wall and ceiling, while VPLs on the pillars and curtains light the floor. The renderings in the bottom row use a higher exposure for illustration. \relax }}{\thepage }}
\newlabel{fig:results:leaks}{{3}{\thepage }}
\newlabel{fig:results:leaks@cref}{{[figure][3][]3}{\thepage }}
\@writefile{lof}{\contentsline {figure}{\numberline {4}{\ignorespaces Light leaks caused by using a relatively large shadow bias with the purpose to hide artifacts of the ISMs. The screenshots to the left use the splat renderer, the right ones the single-pixel renderer. Bottom row uses a higher exposure.\relax }}{\thepage }}
\newlabel{fig:results:bias}{{4}{\thepage }}
\newlabel{fig:results:bias@cref}{{[figure][4][]4}{\thepage }}
\@writefile{toc}{\contentsline {subsection}{\numberline {5.2}Performance}{\thepage }}
\@writefile{lot}{\contentsline {table}{\numberline {1}{\ignorespaces Timings of the ISM renderers with different settings.\relax }}{\thepage }}
\newlabel{tab:results:ism_timings}{{1}{\thepage }}
\newlabel{tab:results:ism_timings@cref}{{[table][1][]1}{\thepage }}
\citation{Cigolle:2014:NormalPacking}
\@writefile{toc}{\contentsline {subsection}{\numberline {5.3}Detailed Performance Measurements for the Single-Pixel Point Renderer}{\thepage }}
\@writefile{lot}{\contentsline {table}{\numberline {2}{\ignorespaces Timing breakdown of the single-pixel point renderer.\relax }}{\thepage }}
\newlabel{tab:results:timing_breakdown_single_pixel}{{2}{\thepage }}
\newlabel{tab:results:timing_breakdown_single_pixel@cref}{{[table][2][]2}{\thepage }}
\@writefile{lot}{\contentsline {table}{\numberline {3}{\ignorespaces Timing breakdown of the pull (PL) and push (PS) phase, and total timings (T). The numbers of the individual steps indicate to which mipmap level they write, which is why the pull phase starts with 1 and the push phase has descending numbers. All timings are in milliseconds.\relax }}{\thepage }}
\newlabel{tab:results:timing_breakdown_pull_push}{{3}{\thepage }}
\newlabel{tab:results:timing_breakdown_pull_push@cref}{{[table][3][]3}{\thepage }}
\@writefile{toc}{\contentsline {subsection}{\numberline {5.4}Memory Usage}{\thepage }}
\@writefile{toc}{\contentsline {subsection}{\numberline {5.5}Comparison of the Splat and Single-Pixel Renderer}{\thepage }}
\@writefile{toc}{\contentsline {section}{\numberline {6}Discussion}{\thepage }}
\bibstyle{abbrv}
\bibdata{../foo-thesis}
\bibcite{barak2013temporally}{1}
\bibcite{Chen:2016:Compactvoxels}{2}
\bibcite{Cigolle:2014:NormalPacking}{3}
\bibcite{Dachsbacher:2005:RSM}{4}
\bibcite{evans2015dreams}{5}
\bibcite{Harris:2007:ParallelReduction}{6}
\bibcite{hedman2016sequential}{7}
\bibcite{hollander2011manylods}{8}
\bibcite{Keller:1997:InstantRadiosity}{9}
\bibcite{Keller:2001:InterleavedSampling}{10}
\bibcite{Laine:2011:SoftwareRasterization}{11}
\bibcite{Marroquim:2007:reconstruction}{12}
\bibcite{Marroquim:2008:reconstruction2}{13}
\bibcite{McGuire2011Data}{14}
\bibcite{novak2011screen}{15}
\bibcite{prutkin2012reflective}{16}
\bibcite{ritschel2011ismsViewAdaptive}{17}
\bibcite{ritschel2008ism}{18}
\bibcite{segovia2006bidirectional}{19}
\bibcite{Strengert:2006:Pyramid}{20}
\bibcite{sugihara2014layered}{21}
\bibcite{sun2015manylightsSVO}{22}
\bibcite{tokuyoshi2015vsgl}{23}
\@writefile{toc}{\contentsline {section}{\numberline {7}Conclusion}{\thepage }}
\@writefile{toc}{\contentsline {section}{\numberline {8}References}{\thepage }}

```
those spaces in line 20 are actually null characters.

**Comment by yitzchak** [comment]
Trying to restart a build in the middle of a current build is probably going cause unexpected behavior no matter what `atom-latex` tries to do. This is connected to #236.

The next version of `atom-latex` will have a process manager which will terminate the previous build if one is still in progress. One may still need to do a clean/rebuild if the previous build was killed.

I'll continue to try to reproduce, but the best that we may be able to do is force a rebuild if the previous build was incomplete.

**Comment by karyon** [comment]
i really don't know what i'm talking about, but is any integration with the build package planned? that would give the users consistent UX (in fact i'd say atom-latex feels a bit alien in atom since it does things quite differently), and maybe make stuff like console output and this process management easier...

**Comment by yitzchak** [comment]
The `build` package does not have sufficient features to support a `latex-build` extension. It takes a very project centric approach much like Visual Studio.

As an aside, in v0.37.1 there should be a "LaTeX Log" button in your status panel which shows errors, so you shouldn't need to look at the console output.

**Comment by karyon** [comment]
but... at least i use latex just like that, with individual projects.

anyhow, i get "An error occurred while trying to run opener (1)" when clicking on that latex log button. should i open a new issue for that?

**Comment by thomasjo** [comment]
> i really don't know what i'm talking about, but is any integration with the build package planned?

We have no plans for any integration with the `build` package. We'd only consider such an integration if it was provided in the core Atom experience (or potentially via a community effort such as https://github.com/atom-community/builder).

**Comment by yitzchak** [comment]
@karyon I have reproduced this so you don't need to open another issue.

**Comment by yitzchak** [comment]
I am closing this since we now have a rebuild command, a process manager and improved clean logic. The best solution for a broken build is a rebuild and if that fails a clean, then a build. Since such actions result in deleting user files we will not trigger unless the logic is flawless.
